### DISCUSSION TITLE
Cors whitelist and SMTP

### DISCUSSION THREAD

**Discussion by xavierperez06** [lpm0073/cookiecutter-openedx-devops#41] [Q&A]
Hello, and hope you are having a great day.

I'm trying to configure AWS SES, and everything seems ok, but when i try to send an email i got this error:

2023-01-18 13:55:03,010 INFO 23 [openedx.core.djangoapps.cors_csrf.helpers] [user 12] [ip REDACTED] helpers.py:64 - Origin 'https://courses.REDACTED' was not in `CORS_ORIGIN_WHITELIST`; full referer was 'https://courses.REDACTED/account/settings' and requested host was 'courses.REDACTED'; CORS_ORIGIN_ALLOW_ALL=False
[pid: 23|app: 0|req: 10/21] REDACTED () {74 vars in 3847 bytes} [Wed Jan 18 13:55:02 2023] POST /password_reset/ => generated 281 bytes in 575 msecs (HTTP/1.1 200) 8 headers in 535 bytes (1 switches on core 0)

I checked the values, and infact, the URL it's not in the config of the LMS pod.
![image](https://user-images.githubusercontent.com/34634037/213265060-d91e8c9e-e1c3-47a9-a857-624c5c8f01d2.png)

But i have them in the _settings_merge.yml_ file.
![image](https://user-images.githubusercontent.com/34634037/213265281-6e0386eb-ec2e-4599-b8bc-9c19960b1498.png)

You have any idea on what is overwritting my config? I cannot find it.

Thanks

**Comment by lpm0073** [comment]
Hi Xavier, confirming that i experienced something similar last week in which my settings value for CORS_ORIGIN_WHITELIST in settings_merge.yml was ultimately ignored by tutor. i got around this (in my case) by adding a line like  'tutor config save -set ....' in the deployment workflow.

**Reply by xavierperez06** [reply]
Awesome, in which step? I ask because i tried this:
![image](https://user-images.githubusercontent.com/34634037/213270893-a2583003-98fc-4173-bb2c-a92081af8b3f.png)
But didn't work...Maybe i'm doing something wrong?

**Reply by xavierperez06** [reply]
@lpm0073 it's weird... I added email configuration in the settings_merge file, and that setting is changed. But with CORS, i can't make it work. And can't test if emails and badges works due CORS not being configured.

**Reply by lpm0073** [reply]
This worked for me: 

<img width="492" alt="Screenshot 2023-01-20 at 16 52 51" src="https://user-images.githubusercontent.com/1178819/213819515-878e237d-26f2-48ff-816b-5c895c1a1372.png">

**Reply by lpm0073** [reply]
same as you, i added this step just before the "Dump tutor config" step

**Reply by xavierperez06** [reply]
Did you posted the correct image? The Disable all MFEs it's the same as mine... I don't see what lines did you added to make CORS works. It's driving me crazy haha

**Comment by lpm0073** [comment]
I've duplicated your results. note this screen shot of the contents of /openedx/config/lms.env.yml in the lms pod file system. 
<img width="685" alt="Screenshot 2023-01-21 at 17 25 46" src="https://user-images.githubusercontent.com/1178819/213893996-c1dd229b-b502-4f42-ac70-06ade521def0.png">

however, the contents of /openedx/edx-platform/envs/tutor/production.py illustrate how the original values from lms.env.yml are getting overwritten. i've traced these three lines to https://github.com/overhangio/tutor-mfe/blob/master/tutormfe/patches/openedx-lms-production-settings#L24

LOGIN_REDIRECT_WHITELIST.append("apps.learning.blend-ed.com")
CORS_ORIGIN_WHITELIST.append("http://apps.learning.blend-ed.com")
CSRF_TRUSTED_ORIGINS.append("apps.learning.blend-ed.com")`

farther down, in the CORS section you can see the origin of our problem, but i haven't found what is overwriting the original values.
CORS_ALLOW_CREDENTIALS = True
CORS_ORIGIN_ALLOW_ALL = False
CORS_ALLOW_INSECURE = True
CORS_ALLOW_HEADERS = corsheaders_default_headers + ('use-jwt-cookie',)
CORS_ORIGIN_WHITELIST = []

I *did* find the following in the tutor source: https://github.com/overhangio/tutor/blob/master/tutor/templates/apps/openedx/settings/partials/common_all.py#L205 but this matches the file contents in edx-platform, so i doubt that this is the cause of the problem.

**Comment by xavierperez06** [comment]
Hhmm... Yes, found the same thing that the content of /openedx/config/lms.env.yml in the lms pod file system is correct, but if i run _./manage.py lms diffsettings --all_ the values are wrong.

Still trying to found the source of the problem. Will let you know if i'm able to find anything.
